avatars/processors/inter_record_range_difference.py

```diff
@@ -0,0 +1,265 @@
+                ]
+            ]
+            .isnull()
+            .values.any()
+        ):
+            raise ValueError(
+                f"Expected no missing values for `id_variable`, `target_start_variable`, `target_end_variable` and `sort_by_variable_name`, got columns with nulls instead"
+            )
+
+        df = df.copy()
+
+        # data need to be sorted
+        df = df.sort_values([self.id_variable, self.sort_by_variable_name])
+
+        # store range value (difference between end and start)
+        df[self.new_range_variable_name] = (
+            df[self.target_end_variable] - df[self.target_start_variable]
+        )
+
+        # store first value of target start variable
+        df[self.new_first_variable_name] = df.groupby(self.id_variable)[
+            self.target_start_variable
+        ].transform("first")
+
+        # add a temp column with the value of the previous line
+        df["__end_tmp__"] = df.groupby(self.id_variable)[
+            self.target_end_variable
+        ].shift()
+
+        # store the difference between current and previous value
+        df[self.new_difference_variable_name] = (
+            df[self.target_start_variable] - df["__end_tmp__"]
+        )
+        df[self.new_difference_variable_name] = df[
+            self.new_difference_variable_name
+        ].fillna(0)
+
+        # remove the tmp column and the original target variables
+        df = df.drop(
+            columns=[
+                self.target_start_variable,
+                self.target_end_variable,
+                "__end_tmp__",
+            ]
+        )
+
+        # sort by index to return processed data in the original order
+        df = df.sort_index()
+
+        return df
+
+    def postprocess(self, source: pd.DataFrame, dest: pd.DataFrame) -> pd.DataFrame:
+        if any(
+            [
+                x not in dest.columns.values
+                for x in [
+                    self.id_variable,
+                    self.new_first_variable_name,
+                    self.new_range_variable_name,
+                    self.new_difference_variable_name,
+                ]
+            ]
+        ):
+            raise ValueError(
+                f"Expected valid variable names for `id_variable`, `new_first_variable_name`, `new_range_variable_name` and `new_difference_variable_name`, got `{self.id_variable}`, `{self.new_first_variable_name}`, `{self.new_range_variable_name}` and `{self.new_difference_variable_name}` instead"
+            )
+
+        if self.sort_by_variable_name not in source.columns.values:
+            raise ValueError(
+                f"Expected a valid `sort_by_variable_name`, got {self.sort_by_variable_name} instead"
+            )
+
+        if (
+            dest[
+                [
+                    self.id_variable,
+                    self.new_first_variable_name,
+                    self.new_range_variable_name,
+                    self.new_difference_variable_name,
```

Comment:
All of these come up multiple times in `preprocess` and `postprocess`. How about creating a `self._variables_that_should_not_be_missing = [ ... ] ` in `__init__()`

Reply:
they are not the same variables in `preprocess` and `postprocess` so I would not put it in `__init__()`. But I have create a `variables_to_check` as suggested in both `preprocess` and `postprocess` to avoid duplicating this list of variables